### PR TITLE
SAK-49953 Profile2: Wicket 6 to Wicket 9

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: profile2/tool/pom.xml [hunk 1]
```diff
@@ -111,7 +111,12 @@
 			<artifactId>json-simple</artifactId>
 			<version>${json.simple.version}</version>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>net.bytebuddy</groupId>
+            <artifactId>byte-buddy</artifactId>
+            <scope>compile</scope>
+        </dependency>
```
Comment (ern) [inline profile2/tool/pom.xml:118]: this lib is already included in shared

**Comment by axxter99** [comment]
FYI: https://sakaiproject.atlassian.net/browse/SAK-45738

**Comment by ottenhoff** [comment]
Thank you! 

I did not test this upgrade while doing it. I just used the IDE and LLMs to get through it so it would build.

Now it opens for me and renders as expected, but saving changes does not work as expected. It seems to make the changes asynchronously..... I'm confused and probably need help.

**Comment by jonespm** [comment]
I'm not sure what's going on here but looks like a lot of changes!

If the LLM can figure out how to upgrade delegated access and convert the LinkTree and TreeTable there it'd be a genius. I just feel that that UI and possibly backend will need to be redone to new components and just feels like a multi-week project.

**Comment by ottenhoff** [comment]
> If the LLM can figure out how to upgrade delegated access and convert the LinkTree and TreeTable there it'd be a genius. I just feel that that UI and possibly backend will need to be redone to new components and just feels like a multi-week project.

Seems like Wicket still supports Tree? My guess it's just one day to figure out.

**Comment by jonespm** [comment]
These were on https://sakaiproject.atlassian.net/issues/SAK-33238

From Wicket 7 on they dropped these tree classes and just have the new AbstractTree that has a NestedTree and TableTree.

> The old Tree component in wicket-extensions is removed.
> It was based on Swing APIs and many people complained about this. Use the new Tree component introduced in Wicket 6.0.0 instead.

I believe that TreeTable needed to be converted to one of these. I'm not sure what LinkTree would migrate to maybe NestedTree. It also needed to implement TreeModelProvider. 

I worked on it in 2017 and didn't finish it and didn't get it to work both because I didn't know how Delegated Access fully worked (especially features like the shopping cart) and I also didn't Wicket enough.

https://github.com/sakaiproject/sakai/compare/master...jonespm:DAC-81-treemodel

Though It _looks_ like someone may have forked this old tree to support the more modern versions of Wicket? I don't remember that being around back then. Maybe if that does work it would allow it to more easily upgrade but ideally these components would be migrated.

https://mvnrepository.com/artifact/org.wicketstuff/wicket15-tree 
https://github.com/wicketstuff/wicket1.5-tree.

**Pull request merged**